src/rules/effects/no-effects-in-providers.ts

```diff
@@ -31,26 +32,46 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
   },
   defaultOptions: [],
   create: (context) => {
-    const effectsInProviders: TSESTree.Identifier[] = []
-    const importedEffectsNames: string[] = []
+    const sourceCode = context.getSourceCode()
+    const effectsInProviders = new Set<TSESTree.Identifier>()
+    const effectsInImports = new Set<string>()
 
     return {
       [effectsInNgModuleProviders](node: TSESTree.Identifier) {
-        effectsInProviders.push(node)
+        effectsInProviders.add(node)
       },
       [effectsInNgModuleImports](node: TSESTree.Identifier) {
-        importedEffectsNames.push(node.name)
+        effectsInImports.add(node.name)
       },
       [`${ngModuleDecorator}:exit`]() {
         for (const effectInProvider of effectsInProviders) {
-          if (importedEffectsNames.includes(effectInProvider.name)) {
-            context.report({
-              node: effectInProvider,
-              messageId,
-            })
+          if (!effectsInImports.has(effectInProvider.name)) {
+            continue
           }
+
+          context.report({
+            node: effectInProvider,
+            messageId,
+            fix: (fixer) => getFixes(sourceCode, fixer, effectInProvider),
+          })
         }
+
+        effectsInImports.clear()
+        effectsInProviders.clear()
       },
     }
   },
 })
+
+function getFixes(
+  sourceCode: Readonly<TSESLint.SourceCode>,
+  fixer: TSESLint.RuleFixer,
+  node: TSESTree.Identifier,
+) {
+  const nextToken = sourceCode.getTokenAfter(node)
+  const isNextTokenComma = nextToken && ASTUtils.isCommaToken(nextToken)
+  return [
+    fixer.remove(node),
+    ...(isNextTokenComma ? [fixer.remove(nextToken)] : []),
+  ]
```

Comment:
I could have done it this way

```ts
return fixer.removeRange([
  node.range[0],
  isNextTokenComma ? nextToken.range[1] : node.range[1],
])
```

... but then possibly unnecessary changes would be made, such as removing comments indiscriminately.